Tensile/Components/Signature.py

```diff
@@ -71,7 +71,11 @@ def getDstValueType(kernel, cov):
     return dstValueType
 
 def getCptValueType(kernel, cov):
-    cptValueType = cptValueTypeDict[kernel["ProblemType"]["DataType"].toNameAbbrev()]
+    if kernel["ProblemType"]["DataType"].isHalf() and kernel["ProblemType"]["HighPrecisionAccumulate"]:
+        cptValueType = cptValueTypeDict[kernel["ProblemType"]["ComputeDataType"].toNameAbbrev()] # for HHS case, this will be handled in HBH kernels.
+    else:
+        cptValueType = cptValueTypeDict[kernel["ProblemType"]["DataType"].toNameAbbrev()]
```

Comment:
Assuming this function is getting the compute data type as the name implies, why is this not always

cptValueType = cptValueTypeDict[kernel["ProblemType"]["ComputeDataType"].toNameAbbrev()]

?

Reply:
Good point. It confused me while I was working on it, too. cptValueType was based on DataType, before my commit, and, initially, I modified the def to define the cptValueType based on computeDataType, (as the name suggests). But I faced some errors later on for other types.

Reply:
At least previously I would assume the compute type would follow OutputDataType from legacy code.

Can anyone else weigh in on this?

Reply:
OK I think I finally understand what's going on here, as I initially had the same thought as Chris.
It looks like this code is from before we made compute data type explicit in Tensile. So instead of just pulling the compute data type directly, this function uses a dict to map to the correct type based on the (input) data type. Before this fix this wasn't a problem, but now we need to map F16 to either F16 or F32 depending on if it's HPA or not.

The new code produces the correct result (returning F32 for F16+HPA), but the way it does it is quite confusing. Short of refactoring this entire file, I would suggest bypassing the dict entirely and just setting cptValueType directly
```c++
if kernel["ProblemType"]["DataType"].isHalf() and kernel["ProblemType"]["HighPrecisionAccumulate"]:
    cptValueType = "F32"
else:
    cptValueType = cptValueTypeDict[kernel["ProblemType"]["DataType"].toNameAbbrev()]